src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -133,18 +133,59 @@ private List<ProgrammingExerciseTask> extractTasks(ProgrammingExercise exercise)
         var testCases = programmingExerciseTestCaseRepository.findByExerciseIdAndActive(exercise.getId(), true);
         while (matcher.find()) {
             var taskName = matcher.group("name");
-            var testCaseNames = matcher.group("tests");
+            var capturedTestCaseNames = matcher.group("tests");
 
             var task = new ProgrammingExerciseTask();
             task.setTaskName(taskName);
             task.setExercise(exercise);
-            String[] testNames = testCaseNames.split(",");
-            for (String testName : testNames) {
-                String finalTestName = testName.trim();
-                testCases.stream().filter(tc -> tc.getTestName().equals(finalTestName)).findFirst().ifPresent(task.getTestCases()::add);
+            var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
+
+            for (String testName : testCaseNames) {
+                testCases.stream().filter(tc -> tc.getTestName().equals(testName)).findFirst().ifPresent(task.getTestCases()::add);
             }
             tasks.add(task);
         }
         return tasks;
     }
+
+    /**
+     * Get the test case names from the captured group by splitting by ',' if there are no unclosed rounded brackets for
+     * the current test case. This respects the fact that parameterized tests may contain commas.
+     * Example: "testInsert(InsertMock, 1),testClass[SortStrategy],testWithBraces()" results in the following list
+     * ["testInsert(InsertMock, 1)", "testClass[SortStrategy]", "testWithBraces()"]
+     * @param capturedTestCaseNames the captured test case names matched from the problem statement
+     * @return test case names
+     */
+    private List<String> extractTestCaseNames(String capturedTestCaseNames) {
+        List<String> testCaseNames = new ArrayList<>();
+        int numberUnclosedRoundedBrackets = 0;
+        StringBuilder currentTestCaseName = new StringBuilder();
+        for (int i = 0; i < capturedTestCaseNames.length(); i++) {
+            char currentChar = capturedTestCaseNames.charAt(i);
+
+            // check potential split
+            if (currentChar == ',' && numberUnclosedRoundedBrackets == 0) {
+                testCaseNames.add(currentTestCaseName.toString());
+                currentTestCaseName = new StringBuilder();
+                continue;
+            }
+
+            // count the numbers of brackets
+            if (currentChar == '(') {
+                numberUnclosedRoundedBrackets++;
+            }
+            else if (currentChar == ')') {
+                numberUnclosedRoundedBrackets--;
+            }
+
+            currentTestCaseName.append(currentChar);
+
+            // when at the end of the captured names, add the current name to the result
+            if (capturedTestCaseNames.length() == i + 1) {
+                testCaseNames.add(currentTestCaseName.toString());
+            }
```

Comment:
Why don't you move this line after the for-loop?

Reply:
If a task in the PS does not contain any test case (empty rounded braces), this would otherwise add a test case with an empty name. Of course, you could also check this before the loop.